store/buildSkyblockCalendar.js

```diff
@@ -0,0 +1,256 @@
+/* eslint-disable no-plusplus */
+const { nth } = require('../util/utility');
+
+const months = [
+  'Early Spring',
+  'Spring',
+  'Late Spring',
+  'Early Summer',
+  'Summer',
+  'Late Summer',
+  'Early Autumn',
+  'Autumn',
+  'Late Autumn',
+  'Early Winter',
+  'Winter',
+  'Late Winter',
+];
+
+const hourMs = 50000;
+const dayMs = 24 * hourMs;
+const monthLength = 31;
+const yearLength = months.length;
+
+const monthMs = monthLength * dayMs;
+const yearMs = yearLength * monthMs;
+
+const yearZero = 1560275700000;
+
+const zooStart = yearZero + yearMs * 66;
+const zooTimeLength = yearMs / 2;
+
+const pets = [
+  'ELEPHANT',
+  'GIRAFFE',
+  'BLUE_WHALE',
+  'TIGER',
+  'LION',
+  'MONKEY',
+];
+
+function getOffset(month, day) {
+  return months.indexOf(month) * monthLength * dayMs + (day - 1) * dayMs;
+}
+
+function getZooPet(time) {
+  const iterations = Math.floor((time - zooStart) / zooTimeLength);
+
+  return pets[iterations % pets.length];
+}
+
+function getJacobEventTimes() {
+  const times = [];
+
+  for (let month = 0; month < 12; month++) {
+    let day = 2;
+    while (day <= 31) {
+      times.push({
+        start: getOffset(months[month], day),
+        end: getOffset(months[month], day),
+      });
+
+      if (day === 30) break;
+      day += 3;
+      if (day > 31) {
+        day %= 31;
+        month++;
+      }
+    }
+  }
+  return times;
+}
+
+const events = [
```

Comment:
I think having enums representing these would make users' code more readable (and help avoid encoding issues), like `BANK_INTEREST`